### PR TITLE
feat(measure): measure in webapp without need for flipper

### PR DESCRIPTION
**Pull request by Almouro**
flashlight-measure-demo

### CODE REVIEW

**Review by niebo26 — APPROVED** [review]

File: packages/measure/src/server/constants.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const PORT = 3000;
+export const WEBAPP_URL =
+  process.env.API_URL === "local"
```
Comment (legkikh) [inline packages/measure/src/server/constants.ts]: localhost in both cases => choose better naming?

File: packages/web-reporter/src/index.html [hunk 2]
```diff
@@ -2,8 +2,8 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <title>Performance Profiler</title>
-    <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%221em%22 font-size=%2290%22>🚀</text></svg>">
+    <title>Flashlight</title>
+    <link rel="icon" href="https://app.flashlight.dev/favicon.ico" />
```
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/web-reporter/src/index.html:6]: image
 yup 👍

File: packages/measure/src/__tests__/measure.test.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import { removeCLIColors } from "./utils/removeCLIColors";
+import { LogLevel, Logger } from "@perf-profiler/logger";
+
+jest.mock("@perf-profiler/shell", () => ({
+  open: jest.fn(),
+}));
+
+Math.random = () => 0.5;
+
+jest.setTimeout(20000);
```
Comment (legkikh) [inline packages/measure/src/__tests__/measure.test.tsx]: 20 seconds, isn't it too long?

File: packages/measure/src/server/constants.ts (hunk 1 above)
Comment (Almouro) [reply packages/measure/src/server/constants.ts]: Changed it for `DEVELOPMENT_MODE`

File: packages/measure/src/__tests__/measure.test.tsx (hunk 3 above)
Comment (Almouro) [reply packages/measure/src/__tests__/measure.test.tsx]: woups, removed it 👍

File: packages/measure/src/webapp/MeasureWebApp.tsx [hunk 4]
```diff
@@ -1,9 +1,53 @@
...
+      <AppBar>
+        <BundleIdSelector
+          autodetect={autodetect}
+          bundleId={bundleId}
+          onChange={setBundleId}
+        />
+        {bundleId ? (
+          <div className="flex flex-row gap-2">
+            <StartButton start={start} stop={stop} isMeasuring={isMeasuring} />
+            <div data-theme={getThemeColorPalette()[1]}>
```
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/measure/src/webapp/MeasureWebApp.tsx:44]: added a comment on top of it!

File: packages/measure/src/webapp/index.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<!-- This file is almost a copy paste from packages/web-reporter/src/index.html -->
```
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/measure/src/webapp/index.html]: in the end, i just removed the comment 👍

File: packages/measure/src/common/useLogSocketEvents.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { Logger } from "@perf-profiler/logger";
+import { useEffect } from "react";
+import type { Socket } from "socket.io";
+import type { Socket as ClientSocket } from "socket.io-client";
+
+export const useLogSocketEvents = (socket: Socket | ClientSocket) => {
+  useEffect(() => {
+    function onAny(event: string, ...args: unknown[]) {
```
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/measure/src/common/useLogSocketEvents.ts]: renamed to `onSocketEvent`

File: packages/measure/src/webapp/components/TextField.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: "inherit",
+  "& .MuiInputBase-input": {
+    padding: 10,
+    // vertical padding + font size from searchIcon
```
[the remark this answers is not in the document]
Comment (Almouro) [reply packages/measure/src/webapp/components/TextField.tsx:32]: agreed, though I just copy pasted what we had in the Flipper plugin 🙈 can we say it's fine for now?

**Review by legkikh — APPROVED** [review]

**Pull request merged**